gui/wxpython/datacatalog/tree.py

```diff
@@ -682,6 +684,20 @@ def OnCreateMapset(self, event):
                    message=_("Unable to create new mapset: %s") % e,
                    showTraceback=False)
 
+    def OnCreateLocation(self, event):
+        """
+        Location wizard started
+        """
+        grassdatabase, location, user_mapset = (
+            create_location_interactively(self,
+                                          self.selected_grassdb[0].data['name'])
+        )
+        if location is not None:
+            self.ReloadTreeItems()
+
+            if user_mapset:
+                self.OnCreateMapset(event)
```

Comment:
It seems to me now that the best way how to do this is to put the `if user_mapset:` part to `create_location_interactively()`. In case loc wizard says a mapset needs to be created, it is appropriate for `create_location_interactively()` to call `create_mapset_interactively()`. No need to involve the caller (i.e., data catalog or startup) into the process.

Reply:
if user_mapset:
user_mapset = **create_mapset_interactively**(guiparent,
                                                  gWizard.grassdatabase,
                                                  gWizard.location)